pyuvdata/uvbeam.py

```diff
@@ -1019,12 +1019,46 @@ def __iadd__(self, other):
         self.__add__(other, inplace=True)
         return self
 
+    def _get_beam(self, stokes):
+        """
+        Get the healpix beam map corresponding to the specififed stokes polarization,
+        'pseudo_I', 'pseudo_Q', etc., or linear dipole polarization, 'XX', 'YY', etc.
+        """
+        pol_array = self.polarization_array
+        if stokes == 'pseudo_I':
+            if 1 in pol_array:
+                stokes_I_ind = np.where(np.isin(pol_array, 1))[0][0]
+                beam = self.data_array[0, 0, stokes_I_ind]
+            elif -5 in pol_array and -6 in pol_array:
+                # Assume A_I = (B_xx + B_yy)/2
+                xx_ind = np.where(np.isin(pol_array, -5))[0][0]
+                yy_ind = np.where(np.isin(pol_array, -6))[0][0]
+                beam = 0.5 * (self.data_array[0, 0, xx_ind] + self.data_array[0, 0, yy_ind])
+            else:
+                raise ValueError('Do not have the right polarization information')
+        elif stokes.upper() == 'XX':
+            if -5 in pol_array:
+                xx_ind = np.where(np.isin(pol_array, -5))[0][0]
+                beam = self.data_array[0, 0, xx_ind]
+            else:
+                raise ValueError('Do not have the right polarization information')
+        elif stokes.upper() == 'YY':
+            if -6 in pol_array:
+                yy_ind = np.where(np.isin(pol_array, -6))[0][0]
+                beam = self.data_array[0, 0, yy_ind]
+            else:
+                raise ValueError('Do not have the right polarization information')
+        else:
+            raise NotImplementedError("Stokes {} not yet implemented...".format(stokes))
+
+        return beam
+
     def get_beam_area(self, stokes='pseudo_I'):
```

Comment:
I would favor renaming the 'stokes' option to something like 'polarization' or 'polarization_name' because 'xx' and 'yy' are not stokes types, they're polarization types.

Reply:
I agree, and I also think that we should not use 'pseudo_I' or 'pseudo_Q' etc. because it is not consistent with the utils.polstr2num convention we have already adopted within `pyuvdata` and within `hera_cal`. Although I appreciate the philosophy behind making the "pseudo" distinction, I think it makes things unnecessarily complicated and not self-consistent. What are your thoughts?

Reply:
hmm... I hear what you're saying but I'm very sensitive to the confusion between stokes and pseudo-stokes. Let me talk to a few other people.